### PR TITLE
Tpetra: FEMultiVector Merge (Take 3)

### PR DESCRIPTION
**Pull request by csiefer2**
Because some people don't want to wait for an autotester repop.

### CODE REVIEW

File: packages/tpetra/core/src/Tpetra_FEMultiVector_decl.hpp [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+
+  template <class Scalar = ::Tpetra::Details::DefaultTypes::scalar_type,
+            class LocalOrdinal = ::Tpetra::Details::DefaultTypes::local_ordinal_type,
+            class GlobalOrdinal = ::Tpetra::Details::DefaultTypes::global_ordinal_type,
+            class Node = ::Tpetra::Details::DefaultTypes::node_type>
+  class FEMultiVector :
+    public MultiVector<Scalar, LocalOrdinal, GlobalOrdinal, Node>
+  {
+  private:
+    friend class MultiVector<Scalar, LocalOrdinal, GlobalOrdinal, Node>;
```
Comment (mhoemmen) [inline packages/tpetra/core/src/Tpetra_FEMultiVector_decl.hpp]: I'm pretty sure we wouldn't need this if we just use `this->` everywhere we call a base class method or refer to a base class thing.

File: packages/tpetra/core/src/Tpetra_FEMultiVector_decl.hpp [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+    /// \param numVectors [in] Number of columns of the resulting MultiVector.
+    // NOTE: A map AND an importer need to be arguments because in serial, the importer will be null
+    // This will default to importer->getTargetMap() being the active MultiVector
+    FEMultiVector(const Teuchos::RCP<const Map<LocalOrdinal,GlobalOrdinal,Node> > & map,
+                  const Teuchos::RCP<const Import<LocalOrdinal,GlobalOrdinal,Node> >& importer,
+                  const size_t numVecs,
+                  const bool zeroOut = true);
+
+    //! Destructor (virtual for memory safety of derived classes).
+    virtual ~FEMultiVector () {delete inactiveMultiVector_;}
```
Comment (mhoemmen) [inline packages/tpetra/core/src/Tpetra_FEMultiVector_decl.hpp]: not fixed yet

File: packages/tpetra/core/src/Tpetra_FEMultiVector_decl.hpp [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+    {
+      FE_ACTIVE_TARGET,
+      FE_ACTIVE_SOURCE
+    };
+
+    // Switches which Multivector is active (without migrating data)
+    void switchActiveMultiVector();
+
+    // This is whichever multivector isn't currently active
+    MultiVector<Scalar, LocalOrdinal, GlobalOrdinal, Node> * inactiveMultiVector_;
```
Comment (mhoemmen) [inline packages/tpetra/core/src/Tpetra_FEMultiVector_decl.hpp]: not fixed yet

File: packages/tpetra/core/src/Tpetra_FEMultiVector_decl.hpp (hunk 1 above)
Comment (csiefer2) [reply packages/tpetra/core/src/Tpetra_FEMultiVector_decl.hpp]: One would think so.  But that's still not true.

**Comment by csiefer2** [comment]
@mhoemmen Trivial to do.  But is it actually useful?

**Comment by csiefer2** [comment]
@mhoemmen Trivial to do.  But is it actually useful?

**Comment by mhoemmen** [comment]
@csiefer2 argh conflicts :(

**Comment by mhoemmen** [comment]
Oops never mind it was just in a weird state before.  Thanks! :D

**Pull request merged**